pkg/controller/usersignup/status.go

```diff
@@ -257,7 +258,38 @@ func (u *statusUpdater) setStatusDeactivationNotificationCreationFailed(userSign
 		})
 }
 
-func (u *statusUpdater) updateStatus(logger logr.Logger, userSignup *toolchainv1alpha1.UserSignup,
+func (u *statusUpdater) setStatusDeactivatingNotificationCreated(userSignup *toolchainv1alpha1.UserSignup, _ string) error {
+	return u.updateStatusConditions(
+		userSignup,
+		toolchainv1alpha1.Condition{
+			Type:   toolchainv1alpha1.UserSignupUserDeactivatingNotificationCreated,
+			Status: corev1.ConditionTrue,
+			Reason: toolchainv1alpha1.UserSignupDeactivatingNotificationCRCreatedReason,
+		})
+}
+
+func (u *statusUpdater) setStatusDeactivatingNotificationUserIsActive(userSignup *toolchainv1alpha1.UserSignup, _ string) error {
+	return u.updateStatusConditions(
+		userSignup,
+		toolchainv1alpha1.Condition{
+			Type:   toolchainv1alpha1.UserSignupUserDeactivatingNotificationCreated,
+			Status: corev1.ConditionFalse,
+			Reason: toolchainv1alpha1.UserSignupDeactivatingNotificationUserIsActiveReason,
```

Comment:
This reason looks a bit confusing to me...
User is still active even if he/she is in the deactivating state until we actually deactivate the account. Isn't he?

Reply:
That's correct, I was wondering if there might be a better reason string to use here, or perhaps just an empty string even.

Reply:
If you don't have a better name (I don't) then let's make sure this reason const has detailed comment explaining what this reason means (in API where we declare the const).

Reply:
I've been trying to think of a better name... let's discuss during our call tomorrow.

Reply:
Resolved as discussed.